pkg/curatedpackages/bundlesource_test.go

```diff
@@ -0,0 +1,91 @@
+package curatedpackages_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/aws/eks-anywhere/pkg/curatedpackages"
+)
+
+func TestBundleSourceSet(t *testing.T) {
+	h := newBSHelper(t)
+	h.Parallel()
+
+	type acceptCase struct {
+		// name of the test
+		name string
+		// input from the user
+		input string
+		// expected result
+		expected curatedpackages.BundleSource
+	}
+
+	accepts := []acceptCase{
+		{"golden path", "cluster", curatedpackages.Cluster},
+		{"case insensitivity", "cLuSTEr", curatedpackages.Cluster},
+		{"golden path", "registry", curatedpackages.Registry},
+		{"case insensitivity #2", "Registry", curatedpackages.Registry},
+		{"whitespace before", " registry", curatedpackages.Registry},
+		{"whitespace after", "registry ", curatedpackages.Registry},
+	}
+	for _, testcase := range accepts {
+		h.runAcceptCase(testcase.name, testcase.input, testcase.expected)
+	}
+
+	type rejectCase struct {
+		// name of the test
+		name string
+		// input from the user
+		input string
+	}
+
+	rejects := []rejectCase{
+		{"empty", ""},
+		{"double dash", "--"},
+		{"exclamation point", "!"},
+		{"something", "something"},
+		{"junk", "junk"},
+		{"kubeVersion", "1.21"},
+		{"random space in the middle", "reg istry"},
+	}
+	for _, testcase := range rejects {
+		h.runRejectCase(testcase.name, testcase.input)
+	}
+}
+
+//
+// Helpers
+//
+
+type bsHelper struct{ *testing.T }
```

Comment:
I will leave it up to you but imo this code is overly _dried out_ to the point that it is difficult to follow

We basically have written our own testing framework for literally two tests. Also, it's not even making the code shorter, it's just abstracting the assertions. Putting this code directly in the table test loop might even make this smaller.